util/tele/span_logger.go

```diff
@@ -165,10 +172,8 @@ func StartSpanWithLogger(
 	}
 
 	lggr := log.FromContext(ctx, kvs...).WithName(spanName)
-	return ctx, &compositeLogger{
-		loggers: []logr.Logger{
-			corrIDLogger(ctx, lggr),
-			&spanLogger{Span: span},
-		},
-	}, endFn
+	return ctx, NewCompositeLogger([]logr.LogSink{
+		corrIDLogger(ctx, lggr),
+		&spanLogSink{Span: span},
```

Comment:
should we use `NewSpanLogger` here?

Reply:
Not part of this pr, but I'm also wondering if creating new log sink each time is a good approach. Maybe we should try to improve it?

Reply:
not NewSpanLogger because that returns a logger but changed it to NewSpanLogSink which returns a log sink